clean_code/pattern_algorithms.py

```diff
@@ -0,0 +1,201 @@
+from random import choice, randint, shuffle
+from string import digits
+from time import time
+from collections import Counter
+from math import factorial
+
+
+def random_string_of_size(n):
+    return ''.join(choice(digits) for i in range(n))
+
+
+def random_binary_string_of_size(n):
+    return ''.join(choice(['0', '1']) for i in range(n))
+
+
+def random_int_list_of_size(n):
+    return list(map(lambda _: randint(0, n), range(n)))
+
+
+def permutations(base_string):
+    pass
+
+
+def print_n_biggest_numbers(numbers, amount=20):
+    if len(numbers) > amount:
+        biggest_numbers = sorted(numbers[:amount], reverse=True)
+    else:
+        return numbers
+
+
+    for number in numbers:
+        if number > biggest_numbers[-1]:
+            add_number_to_sorted_list(biggest_numbers, number)
+            del biggest_numbers[-1]
+
+    print(biggest_numbers)
+
+
+def add_number_to_sorted_list(numbers, number):
+    for index in range(len(numbers)):
```

Comment:
see https://stackoverflow.com/questions/8024571/insert-an-item-into-sorted-list-in-python